src/7-to-8/major-changes/parameters.rst

```diff
@@ -0,0 +1,100 @@
+Parameters
+==========
+
+.. admonition:: does this change affect me?
+   :class: tip
+
+   If you use Cylc parameters with negative offsets (e.g. ``foo<x-1>``).
+
+
+Overview
+--------
+
+There has been a subtle change in the way negative offsets are handled in parameters.
+
+
+Example
+-------
+
+If you have a parameter ``x`` with the values 1, 2 & 3:
+
+.. code-block:: cylc
+
+   [task parameters]
+      x = 1..3
+
+And use it like so:
+
+.. code-block:: cylc-graph
+
+   a<x-1> => b<x> => c<x>
+
+There is some ambiguity about how this should be interpreted when ``x=1``
+because ``<x-1>`` would be ``0`` which is not a valid value for the parameter
+``x``.
+
+Cylc 7 removed the part of the expression which was out of range resulting in a
+partial evaluation of that line:
+
+.. code-block:: cylc-graph
+
+           b_x1 => c_x1  # x=1
+   a_x1 => b_x2 => c_x2  # x=2
+   a_x2 => b_x3 => c_x3  # x=3
+
+Whereas Cylc 8 will remove the entire line:
+
+.. code-block:: cylc-graph
+
+   a_x1 => b_x2 => c_x2  # x=2
+   a_x2 => b_x2 => c_x2  # x=3
```

Comment:
I'm struggling to remember this one it was so long ago.

Is this correct, or does Cylc remove the part of the line after the first out-of-bounds parameter?

Reply:
(you got it right in the other comment).